### ISSUE TITLE
About testing precision on my own dataset

### ISSUE THREAD

**Issue by TonightGo** [wvangansbeke/LaneDetection_End2End#35]
Hi, wvangansbeke:
     I have transplant the pytorch code to caffe and compare their precision. The output of the network is nearly the same. But when I test the network my own dataset use my caffe code, the detection result is not ideal. What  should I pay attention to when I test my own dataset? And what should I do if I want to improve the performance?

**Comment by wvangansbeke** [comment]
Hi @TonightGo,

If you test on you own dataset, you should take into account:
1.  You should calculate your own homography matrix `M` and ground truth parameters `beta`.
2. Change the amount of filters of the last convolution layer in the network to amount of lane lines the network should detect.

I think that should be it.

Kind regards,
Wouter

**Comment by TonightGo** [comment]
Hi @wvangansbeke,

    Thank you for reply very much. But how to use a fixed matrix M between two different datasets? I mean, for example, training on the TuSimple dataset and testing on the CULane dataset.

Kind regards,
Wouter

**Comment by TonightGo** [comment]
Hi @wvangansbeke,

Thank you for reply very much. But how to use a fixed matrix M between two different datasets? 
I mean, for example, training on the TuSimple dataset and testing on the CULane dataset.

Kind regards,
Wouter

**Comment by TonightGo** [comment]
Hi Wouter,

Thank you for reply very much. But how to use a fixed matrix M between two different datasets?
I mean, for example, training on the TuSimple dataset and testing on the CULane dataset.

Kind regards,
ICE

**Comment by TonightGo** [comment]
Hi Wouter,

Thank you for reply very much. But how to use a fixed matrix M between two different datasets?
I mean, for example, training on the TuSimple dataset and testing on the CULane dataset.

Kind regards,
Ice

**Comment by wvangansbeke** [comment]
Hi @TonightGo,

You can just define two fixed matrices. One for TuSimple and one for CULane. The network extracts features from the original perspective after all.

Best regards,
Wouter

**Comment by TonightGo** [comment]
Hi, Wouter,

Thank you for your advice. I will try it.

Best regards,
Ice

**Comment by TonightGo** [comment]
Hi, Wouter,

Thank you for your advice. I will try it. 
And I want to know how to generate the curve parameter labels. I try some methods but can't corresponding to your labels in the file Curve_parameter.json. Wish you complete updating the code this week.

Best regards,
Ice
